File: Chess/src/main/java/game/player/Player.java
package game.player;

import game.move.Move;
import game.board.Board;
import game.piece.*;



public class Player implements java.io.Serializable{
    
    private Team team;
    
    public Player(Team team)
    {
        this.team = team;
    }

    public Team getTeam() {
        return team;
    }

    public void setTeam(Team team) {
        this.team = team;
    }
    
    public void makeMove(Board board, Move move)
    {
        board.getTile(move.getDestinationTile().getCoordinate()).setPiece(move.getCurrentTile().getPiece());
        board.getTile(move.getCurrentTile().getCoordinate()).setPiece(null);
   
    }
}

File: Chess/src/main/java/game/Player.kt
package game

import game.board.Board
import game.piece.Team
import java.io.Serializable

class Player(@JvmField var team: Team) : Serializable {

    fun makeMove(board: Board, move: Move) {
        board.getTile(move.destinationTile.coordinate)?.piece = move.currentTile.piece
        board.getTile(move.currentTile.coordinate)?.piece = null
    }
}